### PR TITLE
Update dependencies

### LINKED ISSUE #226: Update Dependencies Package
Currently, max version for few packages are old.

 device_info_plus: ">=3.2.0 <4.2.0"
package_info_plus: ^1.3.0

As, on time of reporting,
device_info_plus = [6.0.0](https://pub.dev/packages/device_info_plus/versions/6.0.0)
package_info_plus=[2.0.0](https://pub.dev/packages/package_info_plus/versions/2.0.0)

Please update the version of dependency package

### PR DESCRIPTION
**Pull request by Zazo032**
Fixes #226

### CODE REVIEW

**Comment by Zazo032** [comment]
@larryaasen this PR aims to fix the following issue:

```
Because upgrader 4.8.0 depends on package_info_plus ^1.3.0 and no versions of upgrader match >4.8.0 <5.0.0, upgrader ^4.8.0 requires package_info_plus ^1.3.0.
So, because <project> depends on both package_info_plus ^2.0.0 and upgrader ^4.8.0, version solving failed.
pub get failed (1; So, because <project> depends on both package_info_plus ^2.0.0 and upgrader ^4.8.0, version solving failed.)
```

File: pubspec.yaml [hunk 1]
```diff
@@ -12,28 +12,28 @@ dependencies:
     sdk: flutter
 
   # From fluttercommunity.dev: Get current device information from within the Flutter application.
-  device_info_plus: ">=3.2.0 <4.2.0"
+  device_info_plus: ^6.0.0
```
Comment (larryaasen) [inline pubspec.yaml:15]: @Zazo032 Are you suggesting that we drop support for 3.2.0?

Comment (larryaasen) [reply pubspec.yaml:15]: What about supporting ">=4.0.0 <7.0.0"?

File: pubspec.yaml [hunk 2]
```diff
@@ -12,28 +12,28 @@ dependencies:
...
-  url_launcher: ^6.1.0
+  url_launcher: ^6.1.6
 
   # A dart library for comparing and incrementing version numbers with Semantic Versioning.
-  version: ">=2.0.0 <3.1.0"
+  version: ^3.0.2
 
   # A dart library for parsing, traversing, querying, transforming and building XML documents.
-  xml: ">=5.0.2 <7.0.0"
+  xml: ^6.1.0
```
Comment (larryaasen) [inline pubspec.yaml:36]: Why do you want to drop support for 5.0.2?

Comment (Zazo032) [reply pubspec.yaml:36]: Not required, just bumping the minimum from a 20-months old version (might end up being a blocker)

File: pubspec.yaml (hunk 1 above)
Comment (Zazo032) [reply pubspec.yaml:15]: That could work too. Just suggested 6.0.0 because there should be no behaviour/breaking changes to this plugin between v4 and v6.

**Comment by larryaasen** [comment]
@Zazo032  I just created a prerelease version 5.0.0-alpha.1 with all packages updated to the latest version, and requiring Flutter version 3.0.1 or above. Check it out and let me know if it works for you. https://pub.dev/packages/upgrader/versions/5.0.0-alpha.1

**Comment by Zazo032** [comment]
Just tried version `5.0.0-alpha.1` - seems to be fine now, I could update all our dependenies to latest versions, thanks!

Should I close this PR then?

**Comment by larryaasen** [comment]
@Zazo032 Yes please. I have a regular release coming out soon.

**Pull request closed without merging**